src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/OsxIPGlobalProperties.cs

```diff
@@ -0,0 +1,216 @@
+                }
+
+                IPEndPoint remote = new IPEndPoint(remoteIPAddress, (int)nativeInfo.RemoteEndPoint.Port);
+                connectionInformations[i] = new SimpleTcpConnectionInformation(local, remote, state);
+            }
+
+            return connectionInformations;
+        }
+
+        public override IPEndPoint[] GetActiveTcpListeners()
+        {
+            TcpConnectionInformation[] allConnections = GetActiveTcpConnections();
+            return allConnections.Where(tci => tci.State != TcpState.Listen).Select(tci => tci.RemoteEndPoint).ToArray();
+        }
+
+        public unsafe override IPEndPoint[] GetActiveUdpListeners()
+        {
+            int realCount = Interop.Sys.GetEstimatedUdpListenerCount();
+            int infoCount = (int)(realCount * 1.5f);
+            Interop.Sys.IPEndPointInfo* infos = stackalloc Interop.Sys.IPEndPointInfo[infoCount];
+            while (Interop.Sys.GetActiveUdpListeners(infos, &infoCount) == -1)
+            {
+                var newAlloc = stackalloc Interop.Sys.IPEndPointInfo[infoCount];
+                infos = newAlloc;
+            }
+
+            IPEndPoint[] endPoints = new IPEndPoint[infoCount];
+            for (int i = 0; i < infoCount; i++)
+            {
+                Interop.Sys.IPEndPointInfo endPointInfo = infos[i];
+                int port = (int)endPointInfo.Port;
+                IPAddress ipAddress;
+                if (endPointInfo.NumAddressBytes == 0)
+                {
+                    ipAddress = IPAddress.Any;
+                }
+                else
+                {
+                    byte[] bytes = new byte[endPointInfo.NumAddressBytes];
+                    fixed (byte* bytesPtr = bytes)
+                    {
+                        Buffer.MemoryCopy(endPointInfo.AddressBytes, bytesPtr, bytes.Length, bytes.Length);
+                    }
+                    ipAddress = new IPAddress(bytes);
+                }
+
+                endPoints[i] = new IPEndPoint(ipAddress, port);
+            }
+
+            return endPoints;
+        }
+
+        public override IcmpV4Statistics GetIcmpV4Statistics()
+        {
+            return new OsxIcmpV4Statistics();
+        }
+
+        public override IcmpV6Statistics GetIcmpV6Statistics()
+        {
+            return new OsxIcmpV6Statistics();
+        }
+
+        public override IPGlobalStatistics GetIPv4GlobalStatistics()
+        {
+            return new OsxIPv4GlobalStatistics();
+        }
+
+        public override IPGlobalStatistics GetIPv6GlobalStatistics()
+        {
+            // Although there is a 'net.inet6.ip6.stats' sysctl variable, there
+            // is no header for the ip6stat structure and therefore isn't available.
+            throw new PlatformNotSupportedException();
+        }
+
+        public override TcpStatistics GetTcpIPv4Statistics()
+        {
+            // OSX does not provide separated TCP-IPv4 and TCP-IPv6 stats.
```

Comment:
Then shouldn't one of these throw PlatformNotSupported with a detailed text explaining just this? (I vote to leave the IPv6 only.)
I'm thinking of apps that might want to add the stats. If we leave both they'll get 2x the actual stats on OSX.